docs/rules/for-direction.md

```diff
@@ -0,0 +1,23 @@
+# Enforce "for" loop update clause moving the counter in the right direction. (for-direction)
+
+## Rule Details
+
+A for loop with a stop condition that can never be reached, such as one with a counter that moves in the wrong direction, will run infinitely. While there are occasions when an infinite loop is intended, the convention is to construct such loops as while loops. More typically, an infinite for loop is a bug.
```

Comment:
Nitpick: Can you surround "for" and "while" with backticks so that they get formatted as code?

```markdown
A `for` loop with a stop condition ...
```